generic/LeakyReLU.c

```diff
@@ -0,0 +1,65 @@
+#ifndef TH_GENERIC_FILE
+#define TH_GENERIC_FILE "generic/LeakyReLU.c"
+#else
+
+static int nn_(LeakyReLU_updateOutput)(lua_State *L)
+{
+  THTensor *input = luaT_checkudata(L, 2, torch_Tensor);
+  real negval = luaT_getfieldchecknumber(L, 1, "negval");
+  THTensor *output = luaT_getfieldcheckudata(L, 1, "output", torch_Tensor);
+  int inPlace = luaT_getfieldcheckboolean(L, 1, "inplace");
+
+  if (inPlace) {
+    TH_TENSOR_APPLY(real, input,                   \
+                    if (*input_data <= 0) { \
+                      *input_data *=  negval ;           \
+                    });
+    THTensor_(set)(output, input);
+  } else {
+    THTensor_(resizeAs)(output, input);
+    TH_TENSOR_APPLY2(real, output, real, input,                         \
+                     *output_data = (*input_data > 0) ? *input_data : negval * (*input_data););
+
+  }
+
+  return 1;
+}
+
+static int nn_(LeakyReLU_updateGradInput)(lua_State *L)
+{
+  THTensor *input = luaT_checkudata(L, 2, torch_Tensor);
+  THTensor *gradOutput = luaT_checkudata(L, 3, torch_Tensor);
+  real negval = luaT_getfieldchecknumber(L, 1, "negval");
+  THTensor *gradInput = luaT_getfieldcheckudata(L, 1, "gradInput", torch_Tensor);
+  int inPlace = luaT_getfieldcheckboolean(L, 1, "inplace");
+
+  if (inPlace) {
+    TH_TENSOR_APPLY2(real, gradOutput, real, input,    \
+                     if ((*input_data) <= 0) { \
+                       *gradOutput_data = negval;           \

```

Comment:
This should be a multiplication with negval instead a direct assignment (e.g. ``` *gradOutput_data *= negval; ``` )... 

Same is true for ln. 45 which should become
``` 
*gradInput_data = (*input_data > 0) ? *gradOutput_data :  *gradOutput_data * negval; 
```